components/raftstore/src/store/peer_storage.rs

```diff
@@ -2453,47 +2346,22 @@ mod tests {
         exp_res.push(new_entry(7, 8));
         validate_cache(&store, &exp_res);
 
-        let cap = MAX_CACHE_CAPACITY as u64;
-
-        // result overflow
-        entries = (3..=cap).map(|i| new_entry(i + 5, 8)).collect();
-        append_ents(&mut store, &entries);
-        exp_res.remove(0);
-        exp_res.extend_from_slice(&entries);
-        validate_cache(&store, &exp_res);
-
-        // input overflow
-        entries = (0..=cap).map(|i| new_entry(i + cap + 6, 8)).collect();
-        append_ents(&mut store, &entries);
-        exp_res = entries[entries.len() - cap as usize..].to_vec();
-        validate_cache(&store, &exp_res);
-
-        // compact
-        store.compact_to(cap + 10);
-        exp_res = (cap + 10..cap * 2 + 7).map(|i| new_entry(i, 8)).collect();
+        // compact to min(5 + 1, 7)
+        store.cache.persisted = 5;
+        store.compact_to(7);
+        exp_res = vec![new_entry(6, 7), new_entry(7, 8)];
         validate_cache(&store, &exp_res);
 
-        // compact shrink
-        assert!(store.cache.cache.capacity() >= cap as usize);
-        store.compact_to(cap * 2);
-        exp_res = (cap * 2..cap * 2 + 7).map(|i| new_entry(i, 8)).collect();
-        validate_cache(&store, &exp_res);
-        assert!(store.cache.cache.capacity() < cap as usize);
-
-        // append shrink
-        entries = (0..=cap).map(|i| new_entry(i, 8)).collect();
```

Comment:
Should check append will not shrink instead of just removing the case.

Reply:
The `test_storage_cache_entry` and `test_storage_cache_size_change` can meet your requirement.

Reply:
How? For example, if L208~L219 is added back to the file, will these cases fail?

Reply:
It's because the number of appending entries is not big enough. The `MAX_CACHE_CAPACITY` has been removed now.
So what's the number of appending entries in this test? If we use 1024 in test, maybe someone will use 2048 as `MAX_CACHE_CAPACITY` and still can pass this test.